### PR TITLE
check to ensure DOM element has loaded into ref before creating map

### PR DESCRIPTION
**Pull request by JamieMcDonnell**

### CODE REVIEW

File: src/components/GoogleMap.vue [hunk 1]
```diff
@@ -154,55 +155,78 @@ export default defineComponent({
...
-      const loader = new Loader({
-        apiKey: props.apiKey,
-        version: 'weekly',
-        libraries: props.libraries || ['places'],
-        language: props.language,
-        region: props.region,
+    const onLoaded = () => {
+      // eslint-disable-next-line no-undef
+      const { Map } = (api.value = google.maps);
+      console.log('MAPREF:::', mapRef, google)
```
Comment (ksnyde) [inline src/components/GoogleMap.vue:161]: is this meant to be here? Looks like debugging code.

File: src/components/GoogleMap.vue [hunk 2]
```diff
@@ -154,55 +155,78 @@ export default defineComponent({
...
-          },
-        );
-      });
+          })
+        });
+      } else {
+        watch(mapRef, (newMapRef, oldMapRef) => {
+          console.log('no loader watch: ', newMapRef, oldMapRef)
+          if (newMapRef && !oldMapRef) {
+            console.log('ready to run onLoaded')
```
Comment (ksnyde) [inline src/components/GoogleMap.vue:222]: this should be removed

File: src/components/GoogleMap.vue [hunk 3]
```diff
@@ -154,55 +155,78 @@ export default defineComponent({
...
+              onLoaded()
             }
-          },
-        );
-      });
+          })
+        });
+      } else {
+        watch(mapRef, (newMapRef, oldMapRef) => {
+          console.log('no loader watch: ', newMapRef, oldMapRef)
```
Comment (ksnyde) [inline src/components/GoogleMap.vue:220]: this should be removed

File: src/components/GoogleMap.vue [hunk 4]
```diff
@@ -154,55 +155,78 @@ export default defineComponent({
...
-            if (center) {
-              if (!oldCenter || center.lng !== oldCenter.lng || center.lat !== oldCenter.lat) {
-                map.value?.panTo(center);
-              }
+        loader.load().then(() => {
+          console.log('loader loaded')
+          watch(mapRef, (newMapRef, oldMapRef) => {
+            console.log('loader loaded watch: ', newMapRef, oldMapRef)
+            if (newMapRef && !oldMapRef) {
+              console.log('ready to run onLoaded')
```
Comment (ksnyde) [inline src/components/GoogleMap.vue:213]: this should be removed

File: src/components/GoogleMap.vue [hunk 5]
```diff
@@ -154,55 +155,78 @@ export default defineComponent({
...
+        });
+
+        isLoadingAPI.value = true
 
-            if (center) {
-              if (!oldCenter || center.lng !== oldCenter.lng || center.lat !== oldCenter.lat) {
-                map.value?.panTo(center);
-              }
+        loader.load().then(() => {
+          console.log('loader loaded')
```
Comment (ksnyde) [inline src/components/GoogleMap.vue:209]: this should be removed

File: src/components/GoogleMap.vue [hunk 6]
```diff
@@ -154,55 +155,78 @@ export default defineComponent({
...
+          watch(mapRef, (newMapRef, oldMapRef) => {
+            console.log('loader loaded watch: ', newMapRef, oldMapRef)
```
Comment (ksnyde) [inline src/components/GoogleMap.vue:211]: this should be removed

**Review by ksnyde — COMMENTED** [review]
I'm not actively using this plugin atm but on review this looks fine and I assume you've tested this in the browser? I am juggling a bit too much to do testing right now but on visual inspection if you can remove the debugging log lines noted I'll get this merged in and a new **npm** version pushed out.

File: src/components/GoogleMap.vue (hunk 1 above)
Comment (JamieMcDonnell) [reply src/components/GoogleMap.vue]: removed

File: src/components/GoogleMap.vue (hunk 3 above)
Comment (JamieMcDonnell) [reply src/components/GoogleMap.vue]: removed

File: src/components/GoogleMap.vue (hunk 4 above)
Comment (JamieMcDonnell) [reply src/components/GoogleMap.vue]: removed

File: src/components/GoogleMap.vue (hunk 5 above)
Comment (JamieMcDonnell) [reply src/components/GoogleMap.vue]: removed

File: src/components/GoogleMap.vue (hunk 6 above)
Comment (JamieMcDonnell) [reply src/components/GoogleMap.vue]: removed

**Comment by HusamIbrahim** [comment]
Closing this as the target branch has diverged quite a bit.

**Pull request closed without merging**